treeherder/log_parser/utils.py

```diff
@@ -95,11 +95,20 @@ def _retry(e):
                                              job_guid, check_errors)
     except Exception as e:
         client.update_parse_status(project, job_log_url['id'], 'failed')
+        # unrecoverable http error (doesn't exist or permission denied)
         if isinstance(e, urllib2.HTTPError) and e.code in (403, 404):
-            logger.debug("Unable to retrieve log for %s: %s", log_description, e)
-            return
-        logger.error("Failed to download/parse log for %s: %s", log_description, e)
-        _retry(e)
+            logger.debug("Unable to retrieve log for %s: %s", log_description,
+                         e)
+        # possibly recoverable http error (e.g. problems on our end)
+        elif isinstance(e, urllib2.URLError):
+            logger.error("Failed to download log for %s: %s",
+                         log_description, e)
+            _retry(e)
+        # parse error or other unrecoverable error
+        else:
+            logger.error("Failed to download/parse log for %s: %s",
+                         log_description, e)
```

Comment:
I think we need to raise here, otherwise we won't see the exception in New Relic. We deliberately *don't* raise for the HTTP 403/404 case above, but for everything else we should :-)